contracts/modules/andromeda-date-time/src/contract.rs

```diff
@@ -0,0 +1,139 @@
+use andromeda_modules::date_time::{ExecuteMsg, InstantiateMsg, QueryMsg, Timezone};
+use andromeda_std::{
+    ado_base::{InstantiateMsg as BaseInstantiateMsg, MigrateMsg},
+    ado_contract::ADOContract,
+    common::{actions::call_action, context::ExecuteContext, encode_binary},
+    error::ContractError,
+};
+use chrono::{DateTime, Datelike, Timelike, Weekday};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:andromeda-date-time";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let resp = ADOContract::default().instantiate(
+        deps.storage,
+        env,
+        deps.api,
+        &deps.querier,
+        info,
+        BaseInstantiateMsg {
+            ado_type: CONTRACT_NAME.to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            kernel_address: msg.kernel_address,
+            owner: msg.owner,
+        },
+    )?;
+
+    Ok(resp)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let ctx = ExecuteContext::new(deps, info, env);
+    match msg {
+        ExecuteMsg::AMPReceive(pkt) => {
+            ADOContract::default().execute_amp_receive(ctx, pkt, handle_execute)
+        }
+        _ => handle_execute(ctx, msg),
+    }
+}
+
+#[allow(clippy::match_single_binding)]
+fn handle_execute(mut ctx: ExecuteContext, msg: ExecuteMsg) -> Result<Response, ContractError> {
+    let action_response = call_action(
+        &mut ctx.deps,
+        &ctx.info,
+        &ctx.env,
+        &ctx.amp_ctx,
+        msg.as_ref(),
+    )?;
+
+    let res = match msg {
+        _ => ADOContract::default().execute(ctx, msg),
+    }?;
+
+    Ok(res
+        .add_submessages(action_response.messages)
+        .add_attributes(action_response.attributes)
+        .add_events(action_response.events))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<Binary, ContractError> {
+    match msg {
+        QueryMsg::GetDateTime { timezone } => encode_binary(&get_date_time(env, timezone)?),
+        _ => ADOContract::default().query(deps, env, msg),
+    }
+}
+
+pub fn get_date_time(
+    env: Env,
+    timezone: Option<Timezone>,
+) -> Result<GetDateTimeResponse, ContractError> {
+    let timestamp = env.block.time.seconds() as i64;
+    let timezone_i64 = timezone.unwrap_or(Timezone::Utc) as i64;
+    let offset = match timezone_i64.checked_mul(36) {
+        Some(offset) => offset,
+        None => return Err(ContractError::Std(StdError::generic_err("Overflow error"))),
+    };
+    let local_timestamp = match timestamp.checked_add(offset) {
```

Comment:
why not add `?` at the end of this `.checked_add(offset)` instead of that whole `match` statement?

Reply:
resolved